server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/next/ESNextDiskBBQVectorsWriter.java

```diff
@@ -510,6 +510,8 @@ private CentroidGroups buildCentroidGroups(FieldInfo fieldInfo, CentroidSupplier
         // this are small numbers so we run it wih all the centroids.
         final KMeansResult kMeansResult = new HierarchicalKMeans(
             fieldInfo.getVectorDimension(),
+            null,
```

Comment:
nit: It would be nice to avoid these nulls and -1s with a static `HierarchicalKMeans.ofSerial(...)` and `HierarchicalKMeans.ofConcurrent(...)`  methods